src/Microsoft.Azure.WebJobs.Extensions.Kafka/MultipleItemFunctionExecutor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    /// <summary>
+    /// Executes the functions for an specific partition.
+    /// Used for functions that are expecting multiple items at once
+    /// </summary>
+    public class MultipleItemFunctionExecutor<TKey, TValue> : FunctionExecutorBase<TKey, TValue>
+    {
+        public MultipleItemFunctionExecutor(ITriggeredFunctionExecutor executor, IConsumer<TKey, TValue> consumer, CancellationToken cancellationToken, ILogger logger) 
+            : base(executor, consumer, cancellationToken, logger)
+        {
+        }
+
+        protected override async Task ReaderAsync(ChannelReader<KafkaEventData[]> reader, CancellationToken cancellationToken, ILogger logger)
+        {
+            while (await reader.WaitToReadAsync(cancellationToken))
+            {
+                while (reader.TryRead(out var itemsToPublish))
+                {
+                    try
+                    {
+                        // Try to publish them
+                        var triggerInput = KafkaTriggerInput.New(itemsToPublish);
+                        var triggerData = new TriggeredFunctionData
+                        {
+                            TriggerValue = triggerInput,
+                        };
+
+
+                        var functionResult = await this.ExecuteFunctionAsync(triggerData, cancellationToken);
+                        if (functionResult.Succeeded)
```

Comment:
I don’t think we want this check. We want to commit even if execution failed (for now at least - in future may add some retry logic). Event Hubs today also doesn’t make this check

https://github.com/Azure/azure-webjobs-sdk/blob/6930c4ea9ec78db12a05a2d0bf38cce42ab04e13/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs#L176

Reply:
Ok, it makes sense. I will make the required changes.